src/libponyc/ast/parser.c

```diff
@@ -1052,11 +1052,8 @@ DEF(field);
   TOKEN("field name", TK_ID);
   SKIP("mandatory type declaration on field", TK_COLON);
   RULE("field type", type);
-  IF(TK_DELEGATE, RULE("delegated type", provides));
   IF(TK_ASSIGN, RULE("field value", infix));
-  // Order should be:
-  // id type value delegate_type
-  REORDER(0, 1, 3, 2);
+  AST_NODE(TK_NONE); // provides
```

Comment:
That node is unnecessary. I think we can completely remove that child node and refactor the parts of the compiler that try to access it, like we discussed in #1514.